nixos/modules/services/security/bitwaren_rs/default.nix

```diff
@@ -0,0 +1,540 @@
+        The Yubico secret key as registed with at the Yubico server.
+      '';
+    };
+  }
+  // optionalAttrs (packageVersionAtMost "1.6")
+  {
+    localIconExtractor = mkEnableOption "a local favicon extractor instead of Bitwarden's official icon servers";
+  }
+  // optionalAttrs (packageVersionAtLeast "1.6")
+  {
+    iconCacheTTL = mkOption {
+      type = nullOr int;
+      default = null;
+      example = 2592000;
+      description = ''
+        The cache time-to-live (TTL) for successfully obtained icons (in seconds, 0 is forever).
+      '';
+    };
+
+    iconCacheNEGTTL = mkOption {
+      type = nullOr int;
+      default = null;
+      example = 259200;
+      description = ''
+        The cache time-to-live (TTL) for unsuccessfully obtained icons (in seconds, 0 is forever).
+      '';
+    };
+
+    adminToken = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        The authentication token used by the admin page.
+      '';
+    };
+  }
+  // optionalAttrs (packageVersionAtLeast "1.7")
+  {
+    configFile = mkOption {
+      type = nullOr str;
+      default = null;
+      example = "data/config.json";
+      description = ''
+        The JSON file where the user settings will be stored.
+      '';
+    };
+
+    templateFolder = mkOption {
+      type = nullOr str;
+      default = null;
+      example = "/path/to/templates";
+      description = ''
+        The folder path where the templates (e.g. for the emails) will be stored.
+      '';
+    };
+
+    reloadTemplates = mkEnableOption "reloading templates for every request (slow, meant for development)";
+
+    disableIconDownload = mkEnableOption "icon downloading";
+
+    smtpFromName = mkOption {
+      type = nullOr str;
+      default = null;
+      example = "Bitwarden_RS";
+      description = ''
+        The name used in the SMTP From header.
+      '';
+    };
+  }
+  // optionalAttrs (packageVersionAtLeast "1.8")
+  {
+    enableDBWal = mkEnableOption "WAL for the DB";
+
+    iconDownloadTimeout = mkOption {
+      type = nullOr int;
+      default = null;
+      example = 10;
+      description = ''
+        The timeout value when downloading favicons.
+      '';
+    };
+
+    iconBlacklistRegex = mkOption {
+      type = nullOr str;
+      default = null;
+      example = ''192\.168\.1\.[0-9].*^'';
+      description = ''
+        The folder path where the templates (e.g. for the emails) will be stored.
+      '';
+    };
+
+    disable2FARemember = mkOption {
+      type = nullOr bool;
+      default = null;
+      example = false;
+      description = ''
+        Whether to disable 2FA remember.
+        This would force the users to use a second factor to login every time.
+      '';
+    };
+
+    disableAdminToken = mkOption {
+      type = nullOr bool;
+      default = null;
+      example = false;
+      description = ''
+        Whether to disable the use of an admin token.
+      '';
+    };
+  }
```

Comment:
Oh boy, this list of options is a bit too big for my taste. This is exactly the kind of thing I'm hoping to prevent with NixOS/rfcs#42

Reply:
I know (roughly) your position on this, but since I already went through the trouble of mapping all options per version my position on it for now is, as long as there is no consensus on whether to accept the RFC, I am not going to completely redo this module.

Reply:
Then I guess we can agree to disagree, because I don't feel comfortable merging a module like this, I don't want anybody to have to maintain such code in the future, adding new options on every new release.

Reply:
Fair enough. How about I remove the module from the PR for now, but at least already merge the related packages?

Reply:
BTW, I do agree with the general idea behind the RFC. I had the same issues with the module system when I first learned about NixOS. So with regards to the RFC I am in favor of your idea (although I have not read all the details yet), but AFAIK explicit options + extraConfig is still how modules are expected to be arranged, making me uncomfortable to arrange it according to the RFC, since I have yet to see a module arranged like that and the discussion about seems to be ongoing.

Reply:
I'd be fine with merging the package additions. 

The approach in the RFC has been implemented in at least 2 NixOS modules already (one of them by me), see https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md#previous-implementations. I'm trying to get it along as fast as possible, but it's hard because most people don't always have time to review it. I'm holding back from merging PR's that go too strongly against the RFC for the time being, because once introduced, options can't really be removed.

Reply:
Given that people would have to write their own services to use the packages, I thought it would be best to at least try to see how much effort changing it would be. Does the modified module match with what you intended with the RFC? I am not confident about the `enableWebVault`, `enableWebsocket`, and `dataFolder` options, whether they should be options or part of the config option.

Reply:
Oh wow, very nice! Other than a few nitpicks this looks great!